site-content/source/modules/ROOT/pages/community/ask_good_questions.adoc

```diff
@@ -0,0 +1,82 @@
+
+* For **developer questions** including coding or driver issues, https://stackoverflow.com/questions/tagged/cassandra[search on Stack Overflow^].
+* For **admin or ops questions** including installation or maintenance issues, https://dba.stackexchange.com/questions/tagged/cassandra[search on DBA Stack Exchange^].
+* Search the user mailing list archives.
+* Search in the https://the-asf.slack.com/archives/CJZLTM05A[#cassandra channels on ASF Slack^] (requires https://s.apache.org/slack-invite[account signup^]).
+
+Make sure you check the Cassandra logs and application logs because the log entries often give clues.
+
+If you still need help, here are some things to consider when posting your question.
+
+==== Use a meaningful title or subject line
+
+Good: “Cassandra failed to start after upgrading from 3.11.9 to 4.0.1”
+
+Bad: “HELP!!! Cassandra doesn’t start”
+
+==== Explain the problem you're trying to solve
+
+Often users ask if it's good to change parameter X or if they should run command Y but this often isn't helpful. Instead, explain the issue you ran into and what outcome you're after so you'll be more likely to achieve it.
+
+Provide background information. For example, a node went down because of a hardware failure or you were patching the OS. Knowing how you ran into a problem makes it easier to help you.
+
+If you've tried some workarounds based on what you've researched on the internet, include the details in your question. It shows that you have made an effort to look for answers on your own.
+
+==== List software versions
+
+In a lot of cases, the answer depends on the version you're running. Include things like Cassandra version, Java version, driver/framework version (for example, or Java driver 4.11 or Spring Boot 3.3.1).
+
+==== Provide details of the error
+
+Don't just post the error and expect an answer. We often need the full error message and the full stack trace. And in order to decipher the stack trace, make sure you provide the version as above.
+
+==== Provide configuration details
+
+Note how you have configured cassandra.yaml (for example). It isn't necessary to post entire configuration files, just include the properties/items relevant to your question.
+
+Depending on your question, it might be relevant to describe your cluster's topology; number of nodes; whether the nodes are on-premise or in a public cloud; whether they are VMs, pods or bare-metal.
+
+==== Data or app issues
+
+For problematic queries, include the table schema and the full CQL query.
+
+For app-related problems, provide some minimal code that reproduces the issue. It would also help to provide sample data that illustrates the problem.
+
+List the expected result/output versus the actual result/output you got.
+
+'''
+
+==== Are you ready?
+
+For questions about developing apps or need help operating Cassandra, post on the Stack Exchange network. Please do not cross-post the same question to other channels such as Slack or the mailing lists.
```

Comment:
Stack Network

Reply:
> Stack Network

Thanks for the reminder.

> I am not sure we can advise against cross-posting until we have a bot and consolidated pipeline for questions/answers.

In my experience, a lot of users post the same question everywhere (Stack O/E, MLs, community.ds.com, Slack) and I have to point out to them that it's almost always the same people who answer in all those channels. 🙂